pkg/vpcmodel/commonConnectivity.go

```diff
@@ -6,7 +6,139 @@ SPDX-License-Identifier: Apache-2.0
 
 package vpcmodel
 
-import "github.com/np-guard/models/pkg/connection"
+import (
+	"github.com/np-guard/models/pkg/connection"
+	"github.com/np-guard/models/pkg/netp"
+)
+
+// todo: remove stateful from connection.Set (for both options)
+
+// ConnWithStateful connection details
+type ConnWithStateful struct {
+	statefulConn    *connection.Set // stateful TCP connection between <src, dst>
+	nonStatefulConn *connection.Set // nonstateful TCP connection between <src, dst>; complementary of statefulConn
+	otherConn       *connection.Set // non TCP connection (for which stateful is non-relevant)
+	allConn         *connection.Set // entire connection
+}
+
+// operation on ConnWithStateful
+// The operations are performed on the disjoint statefulConn and otherConn and on allConn which contains them;
+// nonStatefulConn - the tcp complementary of statefulConn w.r.t. allConn -
+// is computed as allConn minus (statefulConn union otherConn)
+
+func computeNonStatefulConn(allConn, otherConn, statefulConn *connection.Set) *connection.Set {
+	return allConn.Subtract(otherConn).Subtract(statefulConn)
+}
+
+func EmptyConnWithStateful() *ConnWithStateful {
```

Comment:
add documentation for exported function, here and in the rest of the places later

Reply:
I'll first go over the functions - see which really needs to be exported

Reply:
done